src/lib.rs

```diff
@@ -507,7 +518,26 @@ mod tests {
     }
 
     #[test]
+    fn parse_long_csi_param() {
+        static INPUT: &'static [u8] = &[
+            0x1b, 0x5b, 0x39, 0x32, 0x32, 0x33, 0x33, 0x37, 0x32, 0x30, 0x33, 
+            0x36, 0x38, 0x35, 0x34, 0x37, 0x37, 0x35, 0x38, 0x30, 0x38, 0x6d
+        ];
+
+        let mut dispatcher = CsiDispatcher { params: vec![] };
+
+        let mut parser = Parser::new();
+        for byte in INPUT {
+            parser.advance(&mut dispatcher, *byte);
+        }
+
+        assert_eq!(dispatcher.params[0], &[u16::MAX as i64]);
+
+    }
+    #[test]
     fn parse_osc_with_utf8_arguments() {
+        // INPUT is '\x1b[9223372036854775808m'
+        // The important part is the parameter, which is (i64::MAX + 1)
         static INPUT: &'static [u8] = &[
```

Comment:
Interesting - I had no idea that Rust supported byte literals! I'll make the change.